extensions/import/src/wizard/pages/deployPlanPage.ts

```diff
@@ -0,0 +1,272 @@
+		let currentOperation = '';
+		let dataIssueAlert = false;
+		let currentReportSection: reportSection;
+		let currentTableObj: TableObject;
+
+		// parse the xml deploy plan
+		let report = await this.instance.generateDeployPlan();
+		let p = new parser.Parser({
+			onopentagname(name) {
+				if (name === 'Alert') {
+					currentReportSection = reportSection.Alert;
+				} else if (name === 'Operation') {
+					currentReportSection = reportSection.Operation;
+					currentTableObj = new TableObject();
+				}
+			},
+			onattribute: function (name, value) {
+				if (currentReportSection === reportSection.Alert) {
+					switch (name) {
+						case attributeName.Name: {
+							// only care about showing data loss alerts
+							if (value === 'DataIssue') {
+								dataIssueAlert = true;
+							}
+							break;
+						}
+						case attributeName.Id: {
+							if (dataIssueAlert) {
+								dataLossAlerts.add(value);
+							}
+							break;
+						}
+					}
+				} else if (currentReportSection === reportSection.Operation) {
+					switch (name) {
+						case attributeName.Name: {
+							currentOperation = value;
+							break;
+						}
+						case attributeName.Value: {
+							currentTableObj.object = value;
+							break;
+						}
+						case attributeName.Type: {
+							currentTableObj.type = value;
+							break;
+						}
+						case attributeName.Id: {
+							if (dataLossAlerts.has(value)) {
+								currentTableObj.dataloss = true;
+							}
+							break;
+						}
+					}
+				}
+			},
+			onclosetag: function (name) {
+				// add table entry for the operation item
+				if (name === 'Item') {
+					let isDataLoss = currentTableObj.dataloss ? '⚠️' : '';
+					data.push([isDataLoss, currentOperation, currentTableObj.type, currentTableObj.object]);
+					dataNoDataLoss.push([currentOperation, currentTableObj.type, currentTableObj.object]);
+				}
+			}
+		}, { xmlMode: true, decodeEntities: true });
+
+		p.parseChunk(report);
+
+		if (dataLossAlerts.size > 0) {
+			this.table.updateProperties({
+				data: data,
+				columns: [{
+					value: localize('dacfx.dataLossColumn', 'Data Loss'),
+					width: 50,
+					cssClass: 'center-align',
+					toolTip: localize('dacfx.dataLossTooltip', 'Operations that may result in data loss are marked with a warning sign')
+				},
+				{
+					value: localize('dacfx.operationColumn', 'Operation'),
+					width: 75,
+					toolTip: localize('dacfx.operationTooltip', 'Operation that will occur during deployment')
+				},
+				{
+					value: localize('dacfx.typeColumn', 'Type'),
+					width: 100,
+					toolTip: localize('dacfx.typeTooltip', 'Type of object')
+				},
+				{
+					value: localize('dacfx.objectColumn', 'Object'),
+					width: 300,
+					toolTip: localize('dacfx.objecTooltip', 'Object name')
+				}],
+				width: 875,
+				height: 300
+			});
+
+			this.dataLossCheckbox.enabled = true;
+		} else {
+			this.table.updateProperties({
+				data: dataNoDataLoss,
+				columns: [{
```

Comment:
Rather than duplicating everything, can you put this into an array, and prepend a column for data loss?